web3/contract.py

```diff
@@ -207,13 +156,16 @@ def factory(cls, web3, contract_name=None, **kwargs):
 
         kwargs['web3'] = web3
 
-        for key in kwargs:
+        for key in dict(kwargs):
```

Comment:
why is this wraped in `dict`?

Reply:
I suppose it was a kind of knee-jerk reaction to modifying `kwargs` while looping through it. I don't have a specific reason that I know the assignment mid-loop would break without the change.

Reply:
Ah, I have typically allowed myself to modify `kwargs` since it's not going to leak the same way modifying a normal dictionary would.  I'd suggest at least leaving a comment here, or using one of the `toolz.dicttoolz` functions for the assignment below such as `assoc`

Reply:
I was abstractly worried about modifying the dict mid-iteration. But, I just [RTFA in Python](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects):

> Iterating views while adding or deleting entries in the dictionary may raise a RuntimeError or fail to iterate over all entries.

I take that to mean that replacing existing entries while iterating won't cause any problems.